### PR TITLE
docs: callback alternative to addRemoteTextTrack

### PR DESCRIPTION
**Pull request by kocoten1992**
## Description
Add docs to guide new user to use callback alternative of function ```player.addRemoteTextTrack```

issue: https://github.com/videojs/video.js/issues/4849

### CODE REVIEW

File: docs/guides/text-tracks.md [hunk 1]
```diff
@@ -225,6 +225,14 @@ As mentioned [above](#a-note-on-remote-text-tracks), remote text tracks represen
 * `Player#addRemoteTextTrack(Object options)`
 
   Available options are the same as the [available `track` attributes](#track-attributes). And `language` is a supported option as an alias for the `srclang` attribute - either works here.
+  
+  **Note**: If you need a callback, instead of a callback you could use the technique below:
+  ```js
+  const trackEl = player.addRemoteTextTrack({src: 'en.vtt'}, false);
+  trackEl.on('load', function() {
```
Comment (gkatsev) [inline docs/guides/text-tracks.md]: we should use `addEventListener` because if native tracks are used `trackEl` will reference the native object.

**Comment by gkatsev** [comment]
Also, thanks for getting the ball rolling on this.

**Comment by kocoten1992** [comment]
@gkatsev , after some testing, this is how thing play out:

```player.addTextTrack('kind', 'label', 'lang')``` it fully synchronous, so we don't have to wait for load of any kind, even we add ```load``` it just don't work, example below **don't work**:

```js
var trackEl = player.addTextTrack('captions', 'text1', 'en')

trackEl.addEventListener('load', function () { console.log('pleaseeee work!') }) // never work, because it already loaded - it fully sync
```

Only thing left is ```player.addRemoteTextTrack({src: 'someurl.tld'}, false)```, and since it async, and is the only player in town, should we change ```addEventListener``` to ```on```.

If that oke, I'll do a PR for this, maybe we don't need a section for this, because ```addRemoteTextTrack``` is the only one need callback ?

**Comment by gkatsev** [comment]
Yeah, `addTextTrack` is for programmatic usage, so, there's no need for a `load` event, especially since it returns a TextTrack object and not a TextTrackElement object.

The main thing I think it would be good for adding a "how to wait for tracks to load" section is that it's decoupled from the "API docs" section but yeah, it probably isn't required.
The problem with using `on` is that if you get a native element object it won't have `.on` but you will if you get our emulated element. So, just using `addEventListener` you'll always have it work and it won't break in some cases. Currently, Video.js runs emulated text tracks by default on all browsers except Safari and iOS.

**Pull request merged**